packages/apm/src/integrations/tracing.ts

```diff
@@ -629,6 +648,69 @@ export class Tracing implements Integration {
     // tslint:enable: no-unsafe-any
   }
 
+  /**
+   * Starts tracking the Largest Contentful Paint on the current page.
+   */
+  private static _trackLCP(): void {
+    // Based on reference implementation from https://web.dev/lcp/#measure-lcp-in-javascript.
+
+    // Use a try/catch instead of feature detecting `largest-contentful-paint`
+    // support, since some browsers throw when using the new `type` option.
+    // https://bugs.webkit.org/show_bug.cgi?id=209216
+    try {
+      // Keep track of whether (and when) the page was first hidden, see:
+      // https://github.com/w3c/page-visibility/issues/29
+      // NOTE: ideally this check would be performed in the document <head>
+      // to avoid cases where the visibility state changes before this code runs.
+      let firstHiddenTime = document.visibilityState === 'hidden' ? 0 : Infinity;
+      document.addEventListener(
+        'visibilitychange',
+        event => {
+          firstHiddenTime = Math.min(firstHiddenTime, event.timeStamp);
+        },
+        { once: true },
+      );
+
+      const updateLCP = (entry: PerformanceEntry) => {
+        // Only include an LCP entry if the page wasn't hidden prior to
+        // the entry being dispatched. This typically happens when a page is
+        // loaded in a background tab.
+        if (entry.startTime < firstHiddenTime) {
+          // NOTE: the `startTime` value is a getter that returns the entry's
+          // `renderTime` value, if available, or its `loadTime` value otherwise.
+          // The `renderTime` value may not be available if the element is an image
+          // that's loaded cross-origin without the `Timing-Allow-Origin` header.
+          Tracing._lcp = {
+            // @ts-ignore
+            elementId: entry.id,
+            // @ts-ignore
+            elementSize: entry.size,
+            largestContentfulPaint: entry.startTime,
```

Comment:
There's also `entry.renderTime` and `entry.LoadTime`. 

Unsure if we want to record both values. If `entry.renderTime` is `0`, then this could provide knowledge that the `Timing-Allow-Origin` header was present.

Reply:
> provide knowledge that the `Timing-Allow-Origin` header was present

Talked off of GitHub.

There are three cases, in all cases `entry.startTime` represents the best knowledge about render time, which is what we use to estimate LCP (same used in https://github.com/GoogleChrome/web-vitals/blob/master/src/getLCP.ts)




## Same-domain image

![image](https://user-images.githubusercontent.com/88819/83855739-d1416900-a718-11ea-89ce-dcad37f76394.png)


## Cross-domain image with `Timing-Allow-Origin: *` header

![image](https://user-images.githubusercontent.com/88819/83855700-bf5fc600-a718-11ea-8ccd-4999c028a93d.png)


## Cross-domain image without `Timing-Allow-Origin` header

![image](https://user-images.githubusercontent.com/88819/83855165-fc778880-a717-11ea-9d3f-59e1da1f6e21.png)

Reply:
In the images above, `largestContentfulPaint` is `entry.startTime`.

If `loadTime` by itself or in comparison with `renderTime` is deemed useful in the product, we can add it later. For the purposes of LCP, it is defined as a single number.

Note that since we also get other performance entries and attach to the transaction, the information about loading of every image is already part of the transaction in a span.